skiko/src/iosMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.ios.kt

```diff
@@ -33,15 +32,21 @@ internal class MetalRedrawer(
     private val queue = device.newCommandQueue()!!
     private var currentDrawable: CAMetalDrawableProtocol? = null
     private val metalLayer = MetalLayer()
-
-    init {
-        metalLayer.init(this.layer, contextHandler, device)
-    }
-
-    private val frameDispatcher = FrameDispatcher(SkikoDispatchers.Main) {
+    private val frameListener: NSObject = FrameTickListener {
         if (layer.isShowing()) {
             draw()
         }
+        caDisplayLink.setPaused(true)
```

Comment:
Actually, after looking at this code, will it work with animations?

I see this chain of calls now:

```
draw
  ...
    composeScene.render
      ...
      change state
          need redraw
          caDisplayLink.setPaused(true)

caDisplayLink.setPaused(false)
```
That leads to drawing of only the first frame. It may work with animations like CircularProgressIIndicator (they may dispatch new frames in a coroutine). But may not work with the case, when we change some state in `SideEffect`

Reply:
> when we change some state in SideEffect

probably no in a side effect, but in this case:
```
@Composable
fun Test() {
  val state by remember { mutableStateOf(false) }

  if (state) {
    println("rerendered")
  }
  Canvas(Modifier.size(100.dp)) {
     state = true
  }
}
```

Reply:
CircularProgressIndicator works good:


https://user-images.githubusercontent.com/99798741/222785701-9b84d812-b829-4716-a4af-d8b5fb9ac3e0.mov

Reply:
`@Composable Test()` function prints log only once:
`rerendered`

Reply:
It works how expected. But how does it work, is the chain I described above is wrong🤔. `caDisplayLink.setPaused(false)` can be called during `draw`

Reply:
> caDisplayLink.setPaused(false) can be called during draw

Actually, it's a good comment! We should expect this case. 
A made new commit: move setPause(true) before draw() call. This will allow usage like you described!

Reply:
Looks good!